packages/terminal-extension/src/index.ts

```diff
@@ -108,17 +108,14 @@ function activate(
     });
   }
 
-  // The terminal options from the setting editor.
+  // The cached terminal options from the setting editor.
   let options: Partial<ITerminal.IOptions>;
 
   /**
-   * Update the option values.
+   * Update the cached option values.
    */
   function updateOptions(settings: ISettingRegistry.ISettings): void {
-    options = settings.composite as Partial<ITerminal.IOptions>;
-    Object.keys(options).forEach((key: keyof ITerminal.IOptions) => {
-      ITerminal.defaultOptions[key] = options[key];
-    });
+    options = settings.composite;
```

Comment:
Should the `defaultOptions` be made read only (if they are not already)?

Reply:
What do you think? I would normally think yes, but they were being written to before.

Reply:
I don't know what the reasoning was for writing to them in the first place, but I saw you marking this as a typing error, so I was wondering if this should be cleaned up too.

Reply:
Ah, the typing error had to do with the way the options were set (the `forEach((key: keyof X))` pattern doesn't work so well anymore), not about writing to the default options.

Reply:
If they were really meant to be readonly, we'd probably take them out of the interface file, right?

I wasn't trying to make a decision here about how *all* people use those default options, just how *we* were using them. To prevent scope creep here, and avoid making many design decisions tangential to upgrading to ts 3.5, can we open a separate issue about this (unless there is a clear preference one way or the other)?

(I'm a bit sensitive to scope creep after #6412 ballooned far larger than I anticipated :).

Reply:
> Ah, the typing error had to do with the way the options were set (the forEach((key: keyof X)) pattern doesn't work so well anymore), not about writing to the default options.

I was wondering whether it was a little naive to take the commit message at face value 😉 I was basically curious if this fix was necessary due to the TS upgrade, or an unrelated fix while you were in the neighborhood 😄

Reply:
So...conversation resolved (at least as far as this PR is concerned)?